docs/en/reference/tools.rst

```diff
@@ -27,62 +27,53 @@ Configuration
-or the ``em`` helpers to be defined in order to work correctly.
+All the commands of the Doctrine Console require access to the
+``EntityManager`` or the ``DBAL`` Connection. You have to inject them into
+the console application using so called Helper-Sets. Whenever you invoke
+the Doctrine binary the current folder is searched for the file
+``cli-config.php`` file. This file contains the project specific
+configuration.
 
-Whenever you invoke the Doctrine binary the current folder is searched for a
-``cli-config.php`` file. This file contains the project specific configuration:
+If your application uses the ORM package, the project-specifc
+``cli-config.php`` looks like this:
 
 .. code-block:: php
 
     <?php
-    $helperSet = new \Symfony\Component\Console\Helper\HelperSet(array(
-        'db' => new \Doctrine\DBAL\Tools\Console\Helper\ConnectionHelper($conn)
-    ));
-    $cli->setHelperSet($helperSet);
+    use Doctrine\ORM\Tools\Console\ConsoleRunner;
 
-When dealing with the ORM package, the EntityManagerHelper is
-required:
+    // replace with file to your own project bootstrap
+    require_once 'bootstrap.php';
 
-.. code-block:: php
+    // replace with mechanism to retrieve EntityManager in your app
+    $entityManager = GetEntityManager();
 
-    <?php
-    $helperSet = new \Symfony\Component\Console\Helper\HelperSet(array(
-        'em' => new \Doctrine\ORM\Tools\Console\Helper\EntityManagerHelper($em)
-    ));
-    $cli->setHelperSet($helperSet);
+    return ConsoleRunner::createHelperSet($entityManager);
 
-The HelperSet instance has to be generated in a separate file (i.e.
-``cli-config.php``) that contains typical Doctrine bootstrap code
-and predefines the needed HelperSet attributes mentioned above. A
-sample ``cli-config.php`` file looks as follows:
+If your application uses the `DBAL` connection only, you construct the
+helper set like this:
 
 .. code-block:: php
 
     <?php
-    // cli-config.php
-    require_once 'my_bootstrap.php';
+    use Doctrine\DBAL\Tools\Console as DBALConsole;
+    use Symfony\Component\Console\Helper\HelperSet;
 
-    // Any way to access the EntityManager from  your application
-    $em = GetMyEntityManager();
+    // replace with file to your own project bootstrap
+    require_once 'bootstrap.php';
 
-    $helperSet = new \Symfony\Component\Console\Helper\HelperSet(array(
-        'db' => new \Doctrine\DBAL\Tools\Console\Helper\ConnectionHelper($em->getConnection()),
-        'em' => new \Doctrine\ORM\Tools\Console\Helper\EntityManagerHelper($em)
-    ));
+    // replace with mechanism to retrieve EntityManager in your app
+    $connection = GetDbalConnection();
 
-It is important to define a correct HelperSet that Doctrine binary
-script will ultimately use. The Doctrine Binary will automatically
-find the first instance of HelperSet in the global variable
-namespace and use this.
+    return new HelperSet([
+        'db' => new DBALConsole\Helper\ConnectionHelper($connection),
+    ]);
```

Comment:
I don't see a reason why the ORM docs have a "If your application uses the `DBAL` connection only" part. I think this can be removed from the ORM docs.

Reply:
It's my understanding that migrations can be used with and without the ORM (the latter with some additional code and effort). There is the chapter "[Diffing without the ORM](https://www.doctrine-project.org/projects/doctrine-migrations/en/2.2/reference/generating-migrations.html#diffing-without-the-orm)".

Reply:
Yes and I wouldn't expect "without the ORM" to be found in the ORM documentation. :-)
It is more fitting to find such an information in the Migrations documentation than in ORM and it actually can be found in "[Custom Integration](https://www.doctrine-project.org/projects/doctrine-migrations/en/2.2/reference/custom-integration.html)".